### PR TITLE
[Placement group] Release test

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

This adds a release test. It currently doesn't require to setup the real cluster. 

There are 2 types of tests. Please look at https://docs.google.com/document/d/1ANqiYZR8Y3dWnSDXWu2h2RIT7DwUKzXQeaHrVo__LKY/edit for more details.

1. Measure avg latency of creation & removal.
2. Create 999 placement groups in 3 different tasks. Randomly schedule tasks / actors and remove them. When I scheduled actors, GCS actor management was broken (from resolve actor method which were not properly tested cc @ffbin). 
-> When we run in the real cluster, we can probably scale this to 9999 or even more. 999 pg stress tests took less than 10 seconds.
3. It also tests if there were resource leak after all placement groups are removed.

Also, I already found a bug from the placement group. and added a fix to the PR :)!

There's actually one more bug I discovered (from gcs actor management). I will create an issue once this PR is merged.

## Related issue number

https://github.com/ray-project/ray/issues/9941

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: ci/regression_test/stress_tests/placement_group/test_placement_group.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
+# This is stress test to run placement group.
```
Comment (ericl) [inline ci/regression_test/stress_tests/placement_group/test_placement_group.py:1]: Do we need to add instructions to RELEASE_PROCESS?

File: src/ray/gcs/gcs_server/gcs_placement_group_manager.cc [hunk 2]
```diff
@@ -126,7 +126,9 @@ void GcsPlacementGroupManager::OnPlacementGroupCreationFailed(
   // registered.
   auto state = placement_group->GetState();
   RAY_CHECK(state == rpc::PlacementGroupTableData::RESCHEDULING ||
-            state == rpc::PlacementGroupTableData::PENDING);
+            state == rpc::PlacementGroupTableData::PENDING ||
+            state == rpc::PlacementGroupTableData::REMOVED)
+      << "State: " << state;
```
Comment (ericl) [inline src/ray/gcs/gcs_server/gcs_placement_group_manager.cc:131]: Can we split this in a separate PR (/ have a unit test?)

**Review by ericl — APPROVED** [review]

File: ci/regression_test/stress_tests/placement_group/test_placement_group.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    time.sleep(0.1)
+
+# Scenario 1: Create bunch of placement groups and measure how long it takes.
+total_creating_time = 0
+total_removing_time = 0
+repeat = 1
+total_trial = repeat * num_pg
+bundles = [{"GPU": 1, "pg_custom": 1}] * num_nodes
+
+# Create and remove placement groups
```
Comment (ffbin) [inline ci/regression_test/stress_tests/placement_group/test_placement_group.py:45]:
```suggestion
# Create and remove placement groups.
```

**Review by ffbin — APPROVED** [review]
